lib/commands/actions.js

```diff
@@ -0,0 +1,28 @@
+import androidHelpers from '../android-helpers';
+import log from '../logger';
+
+const swipeStepsPerSec = 28;
+
+let commands = {}, helpers = {}, extensions = {};
+
+commands.swipe = async function (startX, startY, endX, endY, duration, touchCount, elId) {
+  if (startX === 'null') {
+    startX = 0.5;
+  }
+  if (startY === 'null') {
+    startY = 0.5;
+  }
+  let swipeOpts = {startX, startY, endX, endY,
+                   steps: Math.round(duration * swipeStepsPerSec)};
+
+  if (typeof elId !== "undefined" && elId !== null) {
```

Comment:
Again, `hasValue` would be useful here. It is also good to just add the element id to the opts, and then do the command outside of the conditional, so the call is not repeated.

```js
if (util.hasValue(elId)) {
  swipeOpts.elementId = elId;
}
return await this.uiautomator2.jwproxy.command(`/touch/perform`, 'POST', {swipeOpts});
```